curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-a-javascript-promise.english.md

```diff
@@ -0,0 +1,56 @@
+---
+id: 5cdafbb0291309899753167f
+title: Create a JavaScript Promise
+challengeType: 1
+---
+
+## Description
+<section id='description'>
+A promise in JavaScript is exactly what it sounds like. You use it to make a promise to do something, usually asynchronously. When the task completes you either fulfill your promise or fail to do so. <code>Promise</code> is a constructor function, so you need to use the <code>new</code> keyword to create one. It takes a function as its argument with two parameters, <code>resolve</code> and <code>reject</code>. These are methods used to determine the outcome of the promise. The syntax looks like this:
```

Comment:
```suggestion
A promise in JavaScript is exactly what it sounds like - you use it to make a promise to do something, usually asynchronously. When the task completes, you either fulfill your promise or fail to do so. <code>Promise</code> is a constructor function, so you need to use the <code>new</code> keyword to create one. It takes a function, as its argument, with two parameters - <code>resolve</code> and <code>reject</code>. These are methods used to determine the outcome of the promise. The syntax looks like this:
```

Reply:
Why would we want to use hyphens here? That can't be correct grammar/punctuation. I added one of the commas. I didn't think the others were needed. Let me know if you still think they should be added.

Reply:
I added a hyphen here so as to answer what the first sentence means. 

You said "A promise in JavaScript is exactly what it sounds like". So, **what does it sound like**? Without a further explanation of what it sounds like, the sentence is incomplete by itself. The hyphen is intended to complete the sentence, by adding the second sentence to complete "what it sounds like".